src/createItemClass/UpdatePermissiveness.tsx

```diff
@@ -0,0 +1,111 @@
+  PublicKeyInput,
+  RadioInput,
+} from "../components/Wizard/Inputs";
+import { State } from "@raindrops-protocol/raindrops";
+import { useWallet } from "@solana/wallet-adapter-react";
+
+const UpdatePermissiveness = ({ handleSubmit, data }: SubStepProps) => {
+  const { publicKey } = useWallet();
+  const schema = yup.object({
+    updatePermissiveness: PermissivenessInput.validation.default([]),
+    permissivenessToUse: yup.string(),
+    metadataUpdateAuthority: PublicKeyInput.validation,
+  });
+  const parentItemClass : State.Item.ItemClass | undefined = data?.parent_itemclass
+  const parentUpdatePermissiveness =
+    parentItemClass?.itemClassData?.settings?.updatePermissiveness;
+  const parentValue = parentUpdatePermissiveness
+    ? PermissivenessInput.transfromToValue(
+        parentUpdatePermissiveness,
+        State.ChildUpdatePropagationPermissivenessType.UpdatePermissiveness
+      )
+    : undefined;
+  type TValues = yup.InferType<typeof schema>;
+  return (
+    <Formik
+      initialValues={{
+        updatePermissiveness: data?.updatePermissiveness || parentValue,
+        permissivenessToUse: data?.permissivenessToUse,
+        metadataUpdateAuthority: data?.metadataUpdateAuthority || publicKey,
+      }}
+      onSubmit={(values: TValues, actions: FormikHelpers<TValues>) => {
+        actions.setSubmitting(true);
+        handleSubmit({
+          ...values,
+          updatePermissiveness_array:
+            PermissivenessInput.transformFromValueToItemClassPermissiveness(
+              values.updatePermissiveness as (keyof State.AnchorPermissivenessType)[],
+              State.ChildUpdatePropagationPermissivenessType.UpdatePermissiveness,
+              parentItemClass
+            ),
+          permissivenessToUse:
+            values?.updatePermissiveness?.length === 0
+              ? undefined
+              : values?.permissivenessToUse,
+        });
+        actions.setSubmitting(false);
+      }}
+      validationSchema={schema}
+    >
+      {(props) => (
+        <Form onSubmit={props.handleSubmit}>
+          <PermissivenessInput.Formik
+            name="updatePermissiveness"
+            permissivenessType={
+              State.ChildUpdatePropagationPermissivenessType.UpdatePermissiveness
+            }
+            title={`Who can update ${data?.name || "your item"}?`}
+            help="This determines who can make updates to this item"
+            value={
+              props.values
+                ?.updatePermissiveness as (keyof State.AnchorPermissivenessType)[]
+            }
+            parentItemClass={parentItemClass}
+            error={props.errors.updatePermissiveness}
+          />
+          {props.values.updatePermissiveness?.length > 1 && (
+            <RadioInput.Inline
+              name="permissivenessToUse"
+              title="Which permissiveness should be used to create this item class?"
+              help="How do you know what to put here //QUESTION"
```

Comment:
@long-banana how does the user know what to select for the updatePermissivenessToUse? Does it need to be one of the selected update permissivenesses?

Reply:
It should be whatever permissiveness they want to assign for updating the item class config as a whole. Assuming I'm correct in understanding the context this radio is shown in.

Reply:
Here: https://github.com/raindrops-protocol/raindrops/blob/02fb16c4e96a68dd9c456b55701f0fb81f785761/rust/item/src/lib.rs#L354 it seems to be related to the parent? So I guess this should be limited to parent update permissiveness options? Can we leave out for item classes without parents?

Reply:
Yes, we can leave out if there is no parent.

Reply:
Updated in https://github.com/raindrop-studios/creator-ui/pull/8/commits/9fef2d87de08a29eca646ca4b810946bbae8993d